### PR TITLE
Introducing `--list` option to `init-windows`

### LINKED ISSUE #13459: Add functionality for `init-windows` to list available templates (with descriptions?)
There's already code that loads up the list of template objects when init-windows is called, we just need to add a /list arg and if present, dump out the table of templates with their names/descriptions

### PR DESCRIPTION
**Pull request by danielayala94**
## Description
Introducing `--list` option to `init-windows` command.

### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
Displays a list of templates available for usage with `init-windows`.

Resolves #13459

### What
- Added `--list` description, in case developers look for command help.
- Created a function to display template key, name and description in a friendly-readable format.

## Screenshots
https://github.com/user-attachments/assets/ae57779b-c257-4fe7-be5d-0066b65eed6d

## Testing
Ran the cli tests locally, with all tests passing after the change.

## Changelog
Should this change be included in the release notes: yes.

Introduced `--list` option to `init-windows` command.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/13986)

### CODE REVIEW

**Comment by TatianaKapos** [comment]
if we want this in 0.76, we will need to backport it after this is merged in!
Backport instructions here: https://github.com/microsoft/react-native-windows/wiki/Backporting-a-Change

File: packages/@react-native-windows/cli/src/commands/initWindows/initWindows.ts [hunk 1]
```diff
@@ -125,107 +125,123 @@ export class InitWindows {
...
-    const templateConfig = this.templates.get(this.options.template)!;
-
-    // Check if there's a passed-in project name and if it's valid
-    if (this.options.name && !nameHelpers.isValidProjectName(this.options.name)) {
-      throw new CodedError(
-        'InvalidProjectName',
-        `The specified name '${this.options.name}' is not a valid identifier`,
-      );
+    if (this.options.list) {
+      this.printTemplateList();
```
Comment (jonthysell) [inline packages/@react-native-windows/cli/src/commands/initWindows/initWindows.ts:146]: Can you just do a return in the if and avoid re-indenting everything below??

File: packages/@react-native-windows/cli/src/commands/initWindows/initWindows.ts [hunk 2]
```diff
@@ -125,11 +125,28 @@ export class InitWindows {
...
   }
 
+  protected printTemplateList() {
+    if (this.templates.size === 0) {
+      console.log('\nNo templates found.\n');
+      return;
+    }
+
+    for (const [key, value] of this.templates.entries()) {
+      console.log(`\n${key} - ${value.name}\n    ${value.description}`);
```
Comment (jonthysell) [inline packages/@react-native-windows/cli/src/commands/initWindows/initWindows.ts:135]: Can you somehow indicate the default template? There should be a boolean property on the template you can use.

**Review by jonthysell — APPROVED** [review]

File: packages/@react-native-windows/cli/src/commands/initWindows/initWindows.ts (hunk 2 above)
Comment (danielayala94) [reply packages/@react-native-windows/cli/src/commands/initWindows/initWindows.ts:135]: Yes! I got it:

image

**Review by marlenecota — APPROVED** [review]

**Pull request merged**